src/Mod/PartDesign/App/FeaturePipe.cpp

```diff
@@ -303,6 +303,7 @@ App::DocumentObjectExecReturn *Pipe::execute(void)
         AddSubShape.setValue(result);
 
         if(base.IsNull()) {
+            result = refineShapeIfActive(result);

```

Comment:
Can there be a case where refine have an effect, when there is no base object?

Reply:
It will refine the pipe itself, and behave as an additive pipe. It doesn't make sense, but its existing behavior, and I wasn't about to break it.

Reply:
Ok great! I consider the effect that subractive pipe on an empty body a bug. I have checked for it in the Helix code like so:

```
 if(base.IsNull()) {

            if (getAddSubType() == FeatureAddSub::Subtractive)
                return new App::DocumentObjectExecReturn("There is nothing to subtract\n");

```

I will open a PR for that later if you don't do it.